src/main/java/engine/entities/copy/BoardCopier.java

```diff
@@ -10,14 +14,30 @@ public BoardCopier() {}
 
     /**
      * Returns a deep clone of the given Board instance.
+     * The given Board object should contain player objects.
      *
      * @see Board#copy() Board.copy()
      *
-     * @param board the board to be cloned
+     * @param oldBoard the board to be cloned
      * @return a clone of the board
      */
-    public Board createCopy(Board board) {
-        // TODO
-        return null;
+    public Board createCopy(Board oldBoard) {
+        Board newBoard = new Board();
+
```

Comment:
Although the code is not too complicated, I think it's a good idea to leave a little comment on what each block here is doing. It'll save a bit of time in the future in case anything else needs to be changed.